purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1327,7 +1327,7 @@ class Purchases @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE) intern
             val cache = DeviceCache(prefs, apiKey)
 
             return Purchases(
-                context,
+                context.getApplication(),
```

Comment:
we use `context.getApplication` in a few places, could we unify it for clarity under the require? 
in line 1308, 
```kotlin
require(context.applicationContext is Application) { "Needs an application context." }
val application = context.getApplication()
```
and then reuse the variable in the other lines?

Reply:
also, why does the `BillingWrapper.ClientFactory` use `applicationContext` instead of `context.getApplication()`?